src/main/scala/io/iohk/ethereum/blockchain/sync/fast/FastSync.scala

```diff
@@ -171,6 +171,8 @@ class FastSync(
 
       case ResponseReceived(peer, BlockHeaders(blockHeaders), timeTaken) =>
         log.info("*** Received {} block headers in {} ms ***", blockHeaders.size, timeTaken)
+        SyncMetrics.setBlockHeadersDownloadedGauge(timeTaken)
```

Comment:
I'm kinda curious about using a gauge for this (and the other metrics) instead of a timer. A downside compared to the timer is that you lose some of the timings between samples (ie. Grafana pulling data from Prometheus.) With a timer you instead get an average and maximum for each interval, and can extend it with a histogram.

Reply:
When you use a timer you can pass a clear method to it and it will calculate the time for you. This case is not so straightforward.
I forgot to add a screenshot of how the metrics were looking like in Grafana, now I am running mantis already on another branch...

Reply:
So I have transformed the metrics into timers, but the values are calculated the same way. And indeed I am using rates now